com.kms.cura.dal/src/com/kms/cura/dal/database/DoctorUserDatabaseHelper.java

```diff
@@ -210,4 +213,21 @@ protected UserEntity getEntityFromResultSet(ResultSet resultSet) throws SQLExcep
 			doctorSpecialityRS.close();
 		}
 	}
+	public DoctorUserEntity searchDoctor(UserEntity entity) throws ClassNotFoundException, SQLException{
+		DatabaseHelper dbh = null;
+		try{
+			dbh = new DatabaseHelper() {
```

Comment:
instead of creating an anonymous class here, just use the getEntityFromResultSet(resultSet) method straight from this class.